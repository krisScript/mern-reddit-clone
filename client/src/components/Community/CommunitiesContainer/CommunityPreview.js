import React,{useContext} from 'react';
import {Link} from 'react-router-dom'
const CommunityPreview = ({community}) => {
    const {icon,name,subscribers} = community
    return (
      <Link to={`/community/${name}`}>
    <div  className="community-preview">
    <div className="community-preview-container">
    <div className="community-preview-container-icon">
    <img src={`http://localhost:8080/images/${icon}`} alt={`${name}`}/>
    </div>
    </div>
    <div className="community-preview-body">
    <h3 className="community-preview-body-name">c/{name}</h3>
     <h4 className="community-preview-body-subscribers">{subscribers} {subscribers === 1 ? 'subscriber' : 'subscribers'}</h4>
    </div>
    </div>
    </Link>
  );
};
export default CommunityPreview;
