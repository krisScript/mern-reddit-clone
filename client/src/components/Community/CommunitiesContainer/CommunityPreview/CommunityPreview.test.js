import React from 'react';
import { render, fireEvent, waitForElement } from 'react-testing-library';
import CommunityPreview from './CommunityPreview';
import {BrowserRouter} from 'react-router-dom'

describe('<CommunityPreview/>',() => {
    const community = {
        name:'test',
  icon:'default',
  description:"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. ",
  creator: 'test master',
  subscribers:10,
  _id:'1',
    }
  
    
    const { container, rerender,getByTestId,queryByValue ,getByPlaceholderText,getByText } = render (<CommunityPreview community={community}  />,{wrapper:BrowserRouter})
      it('snapshot',() => {
        expect(container).toMatchSnapshot()
    })
    it('should have community name element',  () => {
        const communityName =  getByText(`c/${community.name}`)
        expect(communityName.className).toMatch('community-preview-body-name')
        expect(communityName).toBeTruthy;
      });

      it('should have element image with test id "community-logo"', async () => {
        const communityLogo =  getByTestId(`community-logo`)
        expect(communityLogo).toBeTruthy()
        expect(communityLogo.src).toMatch(`http://localhost:8080/images/${community.icon}`)
        expect(communityLogo.alt).toMatch(community.name)
      });
     
      it('should have element with post.title text',  () => {
        const communitySubscribers =  getByText(`${community.subscribers} subscribers`)
        expect(communitySubscribers.className).toMatch('community-preview-body-subscribers')
        expect(communitySubscribers).toBeTruthy;
      });
      
})

