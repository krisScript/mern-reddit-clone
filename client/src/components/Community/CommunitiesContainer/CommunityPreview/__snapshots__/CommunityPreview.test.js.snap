// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommunityPreview/> snapshot 1`] = `
<div>
  <a
    href="/community/test"
  >
    <div
      class="community-preview"
    >
      <div
        class="community-preview-container"
      >
        <div
          class="community-preview-container-icon"
        >
          <img
            alt="test"
            data-testid="community-logo"
            src="http://localhost:8080/images/default"
          />
        </div>
      </div>
      <div
        class="community-preview-body"
      >
        <h3
          class="community-preview-body-name"
        >
          c/
          test
        </h3>
        <h4
          class="community-preview-body-subscribers"
        >
          10
           
          subscribers
        </h4>
      </div>
    </div>
  </a>
</div>
`;
